kiarash/TS/TS:Generics/src/index.ts

```diff
@@ -0,0 +1,9 @@
+type ArrayElement<T> = T extends string[]
+  ? string
+  : T extends string
+  ? never
+  : unknown;
+
+type EnumLike_v1<T extends Record<string, string | number>> = T[keyof T];
```

Comment:
Same here. You are close to the correct implementation, just a minor change would suffice.